funnel/models/notification_types.py

```diff
@@ -0,0 +1,161 @@
+
+    document_model = Rsvp
+    exclude_actor = False
+    roles = ['owner']
+    exclude_actor = False  # This is a notification to the actor
+
+
+class RegistrationCancellationNotification(ProjectIsParent, Notification):
+    """
+    Notification confirming cancelling registration to a project.
+    """
+
+    __mapper_args__ = {'polymorphic_identity': 'rsvp_no'}
+    category = NOTIFICATION_CATEGORY.PARTICIPANT
+    title = __("When I cancel my registration")
+
+    document_model = Rsvp
+    exclude_actor = False
+    roles = ['owner']
+    exclude_actor = False  # This is a notification to the actor
+
+
+class NewUpdateNotification(ProjectIsParent, Notification):
+    """
+    Notifications of new updates.
+    """
+
+    __mapper_args__ = {'polymorphic_identity': 'update_new'}
+
+    category = NOTIFICATION_CATEGORY.PARTICIPANT
+    title = __("When a project posts an update")
+    description = __(
+        "Typically contains critical information such as video conference links"
+    )
+
+    document_model = Update
+    roles = ['project_crew', 'project_participant']
+    exclude_actor = False  # Send to everyone including the actor
+
+
+class ProposalReceivedNotification(ProfileIsParent, Notification):
+    """
+    Notification to editors of new proposals.
+    """
+
+    __mapper_args__ = {'polymorphic_identity': 'proposal_received'}
+
+    category = NOTIFICATION_CATEGORY.PROJECT_CREW
+    title = __("When my project receives a new proposal")
+
+    document_model = Project
+    fragment_model = Proposal
+    roles = ['project_editor']
+    exclude_actor = True  # Don't notify editor of proposal they submitted
+
+
+class ProposalSubmittedNotification(ProjectIsParent, Notification):
+    """
+    Notification to the proposer on a successful proposal submission.
+    """
+
+    __mapper_args__ = {'polymorphic_identity': 'proposal_submitted'}
+
+    category = NOTIFICATION_CATEGORY.PARTICIPANT
+    title = __("When I submit a proposal")
+
+    document_model = Proposal
+    roles = ['creator']
+    exclude_actor = False  # This notification is for the actor
+
+    # Email is typically fine. Messengers may be too noisy
+    default_email = True
+    default_sms = False
+    default_webpush = False
+    default_telegram = False
+    default_whatsapp = False
+
+
+# --- Notifications with fragments -----------------------------------------------------
+
+
+class ProposalCommentNotification(ProjectIsParent, Notification):
+    """
+    Notification of comments on a proposal.
+    """
+
+    __mapper_args__ = {'polymorphic_identity': 'proposal_comment'}
+
+    category = NOTIFICATION_CATEGORY.PARTICIPANT
+    title = __("When my proposal receives a comment")
+    exclude_actor = True
+
+    document_model = Proposal
+    fragment_model = Comment
+    # Note: These roles must be available on Comment, not Proposal. Roles come from
+    # fragment if present, document if not.
+    roles = ['presenter', 'repliedto_commenter']
```

Comment:
We were planning to migrate the parent column in comment to `in_reply_to_id`. Should this be `replied_to_commenter` or something in that line?

Reply:
`replied_to_commenter` makes sense.